File: line/1.py
def solution(inputString):
    check = [0] * 4
    answer = 0
    flag = True
    for el in inputString :
        if el == "(" :
            check[0] += 1
        elif el == ")" :
            if check[0] > 0:
                check[0] -= 1
                answer += 1
            else :
                flag = False
                break
        elif el == "{" :
            check[1] += 1
        elif el == "}" :
            if check[1] > 0:
                check[1] -= 1
                answer += 1
            else :
                flag = False
                break
        elif el == "[" :
            check[2] += 1
        elif el == "]" :
            if check[2] > 0:
                check[2] -= 1
                answer += 1
            else :
                flag = False
                break
        elif el == "<" :
            check[3] += 1
        elif el == ">" :
            if check[3] > 0:
                check[3] -= 1
                answer += 1
            else :
                flag = False
                break
    if flag == False:
        answer = -1
    elif sum(check) > 0:
        answer = -1
    
    return answer

print(solution("[line [plus]]"))

File: line2022/1.js
const solution = (table, languages, preference) => {
  const jobs = [];
  const scores = [];
  for (const line of table) {
    const [job, ...rest] = line.split(" ");
    jobs.push(job);
    scores.push([...rest].reverse());
  }
  const result = [];
  for (const score of scores) {
    let sum = 0;
    for (let i = 0; i < languages.length; i++) {
      if (score.indexOf(languages[i]) > -1) {
        sum += (score.indexOf(languages[i]) + 1) * preference[i];
      }
    }
    result.push(sum);
  }
  const maxValue = Math.max.apply(null, result);
  const answer = [];
  for (let i = 0; i < result.length; i++) {
    if (result[i] === maxValue) {
      answer.push(jobs[i]);
    }
  }
  return answer.sort()[0];
};
solution(
  ["SI JAVA JAVASCRIPT SQL PYTHON C#", "CONTENTS JAVASCRIPT JAVA PYTHON SQL C++", "HARDWARE C C++ PYTHON JAVA JAVASCRIPT", "PORTAL JAVA JAVASCRIPT PYTHON KOTLIN PHP", "GAME C++ C# JAVASCRIPT C JAVA"],
  ["PYTHON", "C++", "SQL"],
  [7, 5, 5]
);
